shared/src/commonMain/sqldelight/local/TrackingTask.sq

```diff
@@ -0,0 +1,68 @@
+import io.customer.shared.sdk.meta.IdentityType;
+import io.customer.shared.tracking.api.model.TrackingErrorReason;
+import io.customer.shared.tracking.constant.QueueTaskStatus;
+import kotlinx.datetime.Instant;
+
+-- TABLE CREATIONS
+-- Any updates in the table should only be done with proper migrations
+
+CREATE TABLE IF NOT EXISTS trackingTask (
+  uuid TEXT NOT NULL,
+  siteId TEXT NOT NULL,
+  type TEXT NOT NULL,
+  createdAt INTEGER AS Instant NOT NULL,
+  updatedAt INTEGER AS Instant NOT NULL,
+  -- time after which the event may be removed without sending to the server
+  expiresAt INTEGER AS Instant DEFAULT NULL,
+  -- time after which the event can be deleted locally, this can be helpful in keeping logs for
+  -- remote debugging in future
+  stalesAt INTEGER AS Instant DEFAULT NULL,
+  identity TEXT,
+  identityType TEXT AS IdentityType,
+  activityJson TEXT NOT NULL,
+  activityModelVersion INTEGER,
+  queueTaskStatus TEXT AS QueueTaskStatus NOT NULL,
+  priority INTEGER NOT NULL DEFAULT 0,
+  retryCount INTEGER NOT NULL DEFAULT 0,
+  statusCode INTEGER DEFAULT NULL,
+  errorReason TEXT AS TrackingErrorReason DEFAULT NULL,
+  PRIMARY KEY (uuid)
+);
```

Comment:
This probably is the final DB design from me. The only thing unavailable from current queue implementation here will be the the Task X being blocked by Task Y. We can handle tasks blocking till identify event still, but I can't think of any other scenario where blocking would be needed. Do you think we still need that in first phase? If yes, what would be your suggestion to handle it?